src/mainClass.js

```diff
@@ -1,18 +1,24 @@
-import { getData, getAllCategories } from "./source";
+import { getData, getCategories } from "./source";
 
 export class AllItems {
 
   constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
+    // categories, search functionality
     this.categories = [];
     this.onReceivedCategories = searchCategory;
-    this.displayChosenLinks = displayLinksForChosenCategory;
+    // links, search functionality and initial display when app starts
     this.links = [];
     this.onReceivedLinks = searchLinks;
+    this.displayChosenLinks = displayLinksForChosenCategory;
+    // localStorage
     this.storageLinks = JSON.parse(localStorage.getItem('links'));
     this.storageCategories = JSON.parse(localStorage.getItem('categories'));
+    // used for search
```

Comment:
Make sure, that you accomplished workflow, that we defined with receive data method, receive callback, getAll and filter text. For both categories and links. So to we have simplified handlings.
So UI could  initialise load of data (for links with/or without category, for categories always same without param).
When user search data text is set as property, and called getAll to receive filtered array.
When clicked on category is called method with param to receive links again (same method as initial load), and when callback called (data received) is called getAll which will return links for rendering. 
Same when filter text changed, we set text prop, and call getAll again to receive property.
So basically we use method to load data, callback to notify ui that data is received and ui could call getAll and render. When we filter by text we set prop text and call getAll to receive data.